RSDKv5/RSDK/Input/Wii/WiiInputDevice.cpp

```diff
@@ -6,20 +6,61 @@ void RSDK::SKU::InputDeviceWii::UpdateInput() {
     WPAD_ScanPads();
 
     this->buttonMasks = WPAD_ButtonsHeld(0);
-
-    this->stateUp     = (this->buttonMasks & WPAD_BUTTON_RIGHT) != 0;
-    this->stateDown   = (this->buttonMasks & WPAD_BUTTON_LEFT) != 0;
-    this->stateLeft   = (this->buttonMasks & WPAD_BUTTON_UP) != 0;
-    this->stateRight  = (this->buttonMasks & WPAD_BUTTON_DOWN) != 0;
-    this->stateA      = (this->buttonMasks & WPAD_BUTTON_2) != 0;
-    this->stateB      = (this->buttonMasks & WPAD_BUTTON_1) != 0;
-    this->stateC      = (this->buttonMasks & 0) != 0;
-    this->stateX      = (this->buttonMasks & WPAD_BUTTON_A) != 0;
-    this->stateY      = (this->buttonMasks & WPAD_BUTTON_B) != 0;
-    this->stateZ      = (this->buttonMasks & 0) != 0;
-    this->stateStart  = (this->buttonMasks & WPAD_BUTTON_PLUS) != 0;
-    this->stateSelect = (this->buttonMasks & WPAD_BUTTON_MINUS) != 0;
-
+    int type = data->exp.type;
```

Comment:
Missing `WPAD_Data()` ?
```suggestion
    WPADData *data = WPAD_Data(0);
    int type = data->exp.type;
```